### PR TITLE
Fixed error message for duplicate vocabulary

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
Hi,

In this PR I fixed a error message, actually I customize error message in case we add duplicate vocabulary.

https://github.com/mitodl/lore/issues/286

@pdpinch @pwilkins @carsongee 

screen shot 2015-07-07 at 2 57 11 pm

### CODE REVIEW

**Comment by amir-qayyum-khan** [comment]
Fix === related jshint issue
 @carsongee

File: ui/static/ui/js/manage_taxonomies.jsx [hunk 1]
```diff
@@ -150,9 +150,18 @@ define('setup_manage_taxonomies', ['reactaddons', 'lodash', 'jquery'],
         url: API_ROOT_VOCAB_URL,
         data: vocabularyData
       }).fail(function(data) {
-        thiz.setState({
-          errorMessage: 'There was a problem adding the Vocabulary'
-        });
+        data = JSON.parse(data.responseText);
```
Comment (noisecapella) [inline ui/static/ui/js/manage_taxonomies.jsx]: Why not use `data.responseJSON` here?

File: ui/static/ui/js/manage_taxonomies.jsx [hunk 2]
```diff
@@ -150,9 +150,18 @@ define('setup_manage_taxonomies', ['reactaddons', 'lodash', 'jquery'],
...
+        if (data.non_field_errors && data.non_field_errors.length > 0) {
+          if (data.non_field_errors[0] === 'The fields repository, name must make a unique set.') {
+            thiz.setState({
+              errorMessage: 'A Vocabulary named already exists. Please choose a different name.'
```
Comment (noisecapella) [inline ui/static/ui/js/manage_taxonomies.jsx]: Use the name of the Vocabulary from the state in the error message.

File: ui/static/ui/js/manage_taxonomies.jsx [hunk 3]
```diff
@@ -150,9 +150,18 @@ define('setup_manage_taxonomies', ['reactaddons', 'lodash', 'jquery'],
         url: API_ROOT_VOCAB_URL,
         data: vocabularyData
       }).fail(function(data) {
-        thiz.setState({
-          errorMessage: 'There was a problem adding the Vocabulary'
-        });
+        data = JSON.parse(data.responseText);
+        if (data.non_field_errors && data.non_field_errors.length > 0) {
+          if (data.non_field_errors[0] === 'The fields repository, name must make a unique set.') {
```
Comment (noisecapella) [inline ui/static/ui/js/manage_taxonomies.jsx]: What if the error message is not the first error in `non_field_errors`?

**Comment by noisecapella** [comment]
In the future please submit your PRs as a branch on mitodl/lore, not as a fork. For example `feature/aqk/fix_duplicate_error_message`

**Comment by pwilkins** [comment]
@noisecapella Amir has "ready-only" access to our LORE repository so forking is his only option.

**Comment by pwilkins** [comment]
Please make your commit message past tense and correct the spelling of "duplicate": 
`Fixed error message for duplicate vocabulary`

**Comment by amir-qayyum-khan** [comment]
Done with fix @noisecapella 

CC @pwilkins

**Comment by noisecapella** [comment]
Looks good, thanks for your work.

**Pull request merged**